Sources/NIOExtras/LengthFieldBasedFrameDecoder.swift

```diff
@@ -0,0 +1,170 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftNIO open source project
+//
+// Copyright (c) 2017-2018 Apple Inc. and the SwiftNIO project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftNIO project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import NIO
+
+///
+/// An enumeration to describe the length of a piece of data in bytes.
+/// It is contained to lengths that can be converted to integer types.
+///
+public enum ByteLength {
```

Comment:
given that this is a public type, mind putting this inside of your decoder? Otherwise it pollutes the `NIOExtras` namespace with this type. Anyway I think `LengthFieldBasedFrameDecoder.ByteLength` is a better name

Reply:
Sure, done.